topdown/eval_test.go

```diff
@@ -349,6 +349,131 @@ func TestTopdownVirtualCache(t *testing.T) {
 			miss:  1,
 			exp:   true,
 		},
+		{
+			note: "partial object: simple, general ref, multiple vars",
+			module: `package p
+			s.t[u].v[w] = true { x = ["foo", "bar"]; u = x[_]; y = ["do", "re"]; w = y[_] }`,
+			query: `data.p.s.t = x; data.p.s.t`,
+			hit:   1,
+			miss:  1,
+			exp: map[string]interface{}{
+				"foo": map[string]interface{}{
+					"v": map[string]interface{}{
+						"do": true,
+						"re": true,
+					},
+				},
+				"bar": map[string]interface{}{
+					"v": map[string]interface{}{
+						"do": true,
+						"re": true,
+					},
+				},
+			},
+		},
+		{
+			note: "partial object: simple, general ref, multiple vars (2)",
+			module: `package p
+			s.t[u].v[w] = true { x = ["foo", "bar"]; u = x[_]; y = ["do", "re"]; w = y[_] }`,
+			query: `data.p.s.t.foo = x; data.p.s.t["foo"]`,
```

Comment:
Is the `["foo"]` significant here? Would it be different if the query was
```
data.p.s.t.foo = x; data.p.s.t.foo
```
?

Reply:
No, that'd be the exact same query. I just used the two different styles to cover all bases.